libraries/streaming-e2e-tests/tests/streaming-e2e.test.js

```diff
@@ -37,90 +37,86 @@ function createDriver(browser) {
     preferences.setLevel(logging.Type.BROWSER, logging.Level.SEVERE);
     options.setLoggingPrefs(preferences);
 
-    const builder = new Builder()
-        .setChromeOptions(options)
-        .forBrowser(browser)
-        .build();
+    const builder = new Builder().setChromeOptions(options).forBrowser(browser).build();
 
-  return builder;
+    return builder;
 }
 
 async function echoMessageInBrowser(driver) {
-  try {
     console.log(`Navigating to "${reactAppEndpoint}"...`);
     await driver.get(reactAppEndpoint);
 
     await ensureNoBrowserErrors(driver);
 
-    console.log('Sleeping to allow Web Chat to load...');
-    await driver.sleep(7000);
+    console.log('Waiting for webchat to load or error is reported...');
+    await Promise.race([
+        driver
+            .wait(until.elementLocated(By.id('webchat-error')))
+            .getText()
+            .then((message) => {
+                throw new Error(`Found #webchat-error DOM element: ${message}`);
+            }),
+        driver.wait(until.elementLocated(By.id('react-webchat'))),
+    ]);
```

Comment:
Great refactoring! This should reduce the amount of test failures by a lot 👍🏼

Reply:
It also reports token fetch errors 😄

Reply:
🥳 :shipit: